src/zarr/api/asynchronous.py

```diff
@@ -194,6 +194,7 @@ async def open(
     zarr_version: ZarrFormat | None = None,  # deprecated
     zarr_format: ZarrFormat | None = None,
     path: str | None = None,
+    storage_options: dict[str, Any] | None = None,
```

Comment:
Worth adding a warning if `storage_opitions is not None`, but an fsspec URL isn't passed (ie. they aren't used)?

Reply:
IMO, an exception in this case is probably best. I'll do that.